java/src/main/kotlin/org/digma/intellij/plugin/idea/runcfg/OTELJarProvider.kt

```diff
@@ -43,31 +45,61 @@ class OTELJarProvider {
     }
 
 
-    fun getOtelAgentJarPath(project: Project): String? {
+    fun getOtelAgentJarPath(configuration: RunConfigurationBase<*>): String? {
         ensureFilesExist()
         val otelJar = getOtelAgentJar()
-         if (otelJar.exists()) {
-            if(SystemInfo.isWindows && project.basePath?.startsWith("//wsl$/") == true){
-                // Converting From: C:\Users\asafc\AppData\Local\Temp\digma-otel-jars\opentelemetry-javaagent.jar
-                // To:              /mnt/c/Users/XXXXX/AppData/Local/Temp/digma-otel-jars/opentelemetry-javaagent.jar
-                val driveLetter = otelJar.absolutePath[0].lowercase()
-                return "/mnt/" + driveLetter + otelJar.absolutePath.substring(2).replace("\\", "/")
+        if (otelJar.exists()) {
+            if (isWsl(configuration)) {
+                return convertWinToWslPath(otelJar.absolutePath)
             }
             return otelJar.absolutePath
         }
 
         return null
     }
 
+    private fun convertWinToWslPath(path: String): String{
+        // Converting From: C:\Users\XXXXX\AppData\Local\Temp\digma-otel-jars\opentelemetry-javaagent.jar
+        // To:              /mnt/c/Users/XXXXX/AppData/Local/Temp/digma-otel-jars/opentelemetry-javaagent.jar
+        val driveLetter = path[0].lowercase()
+        return "/mnt/" + driveLetter + path.substring(2).replace("\\", "/")
+    }
+
+    private fun isWsl(configuration: RunConfigurationBase<*>): Boolean {
+        if (!SystemInfo.isWindows)
+            return false
+
+        val targets = TargetEnvironmentsManager.getInstance(configuration.project).targets.resolvedConfigs()
+        val targetName = (configuration.state as? RunConfigurationOptions)?.remoteTarget
+        if (targetName == null)
+            return false
+
+        val target = targets.firstOrNull { it.displayName == targetName }
+        if (target == null)
+            return false
+
+        if (target !is WslTargetEnvironmentConfiguration)
+            return false
+
+        return true
+    }
+
     private fun getOtelAgentJar(): File {
         return File(downloadDir, OTEL_AGENT_JAR_NAME)
     }
 
 
-    fun getDigmaAgentExtensionJarPath(): String? {
+    fun getDigmaAgentExtensionJarPath(configuration: RunConfigurationBase<*>): String? {
```

Comment:
i think that's not the right place to modify the path.
this one simply accesses the JAR. later on need to take a look at the RunConfigType